server/analytics/content_publishing_report/content_publishing_report.py

```diff
@@ -167,26 +175,57 @@ def get_aggregation_buckets(self, docs, aggregation_ids=None):
         return super().get_aggregation_buckets(docs, aggregation_ids)
 
     def get_custom_aggs_query(self, query, aggs):
-        field = self.parse_field_param(aggs.get("parent", {}).get("terms", {}).get("field"))
-        if field:
-            # Construct the nested aggregation query
+        # reterive parent field
+        parent_field = self.parse_field_param(aggs.get("parent", {}).get("terms", {}).get("field"))
+
+        # reterive child field
+        child = self.get_child_aggs(query)
+        child_field = self.parse_field_param(child.get("child", {}).get("terms", {}).get("field")) if child else None
+
+        # helper function
+        def construct_nested_query(field):
+            return {
+                "nested": {"path": "subject"},
+                "aggs": {
+                    "qcode_filter": {
+                        "filter": {"term": {"subject.scheme": field}},
+                        "aggs": {"qcode_terms": {"terms": {"field": "subject.qcode", "size": 1000}}},
```

Comment:
I wouldn't hardcode `1000` here, would be better to have some control via settings

Reply:
yep